great_expectations/core/configuration.py

```diff
@@ -1,10 +1,24 @@
 """Contains general abstract or base classes used across configuration objects."""
 from abc import ABC
+from collections import Mapping
 from typing import Optional
 
+try:
+    from typing import Protocol
+except ImportError:
+    from typing_extensions import Protocol  # type: ignore[misc]
+
 from great_expectations.types import SerializableDictDot
 
 
+class Loadable(Protocol):
+    """TODO: also be dumpable?"""
+
+    def load(self, x) -> Mapping:
+        """Foo."""
```

Comment:
Might be nice to not force docstring for Protocol methods. But I think tweaking our docstring checker to recognize this is probably more trouble than it's worth.

Reply:
Yup don't let that be a hindrance if not applicable - you can always just bump up the number of errors.